nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-processors/src/main/java/org/apache/nifi/processors/elasticsearch/ScrollElasticsearchHttp.java

```diff
@@ -0,0 +1,415 @@
+import org.codehaus.jackson.JsonNode;
+
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
+@InputRequirement(InputRequirement.Requirement.INPUT_FORBIDDEN)
+@EventDriven
+@SupportsBatching
+@Tags({ "elasticsearch", "query", "scroll", "read", "get", "http" })
+@CapabilityDescription("Scrolls through an Elasticsearch query using the specified connection properties. "
+        + "This processor is intended to be run on the primary node, and is designed for scrolling through "
+        + "huge result sets, as in the case of a reindex.  The state must be cleared before another query "
+        + "can be run.  Each page of results is returned, wrapped in a JSON object like so: { \"hits\" : [ <doc1>, <doc2>, <docn> ] }.  "
+        + "Note that the full body of each page of documents will be read into memory before being "
+        + "written to a Flow File for transfer.")
+@WritesAttributes({
+        @WritesAttribute(attribute = "es.index", description = "The Elasticsearch index containing the document"),
+        @WritesAttribute(attribute = "es.type", description = "The Elasticsearch document type") })
+@Stateful(description = "After each successful scroll page, the latest scroll_id is persisted in scrollId as input for the next scroll call.  "
+        + "Once the entire query is complete, finishedQuery state will be set to true, and the processor will not execute unless this is cleared.", scopes = { Scope.LOCAL })
+public class ScrollElasticsearchHttp extends AbstractElasticsearchHttpProcessor {
+
+    private static final String FINISHED_QUERY_STATE = "finishedQuery";
+    private static final String SCROLL_ID_STATE = "scrollId";
+    private static final String FIELD_INCLUDE_QUERY_PARAM = "_source_include";
+    private static final String QUERY_QUERY_PARAM = "q";
+    private static final String SORT_QUERY_PARAM = "sort";
+    private static final String SCROLL_QUERY_PARAM = "scroll";
+    private static final String SCROLL_ID_QUERY_PARAM = "scroll_id";
+    private static final String SIZE_QUERY_PARAM = "size";
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description(
+                    "All FlowFiles that are read from Elasticsearch are routed to this relationship.")
+            .build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description(
+                    "All FlowFiles that cannot be read from Elasticsearch are routed to this relationship. Note that only incoming "
+                            + "flow files will be routed to failure.").build();
+
+    public static final PropertyDescriptor QUERY = new PropertyDescriptor.Builder()
+            .name("scroll-es-query").displayName("Query")
+            .description("The Lucene-style query to run against ElasticSearch").required(true)
```

Comment:
Interesting, yeah by default I believe string field should be analyzed.  I'm not an ES expert, so it could be something in the configuration.  I think the important thing is that the NiFi processor behavior matches the ES REST API behavior, though.